### PR TITLE
Feature new release section

### PR DESCRIPTION
**Pull request by smabhena**
Added in new releases section to home page

Screenshot 2022-04-06 at 14 06 34

### CODE REVIEW

File: Mooov/MooovTests/HomeTests/HomeTests.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class HomeTests: XCTestCase {
+    private var viewModel: HomePageViewModel!
+    private weak var mockDelegate: MockDelegate!
+    private var mockRepository: MockRepository!
+
+    override func setUp() {
+        super.setUp()
+        mockRepository = MockRepository()
+        viewModel = HomePageViewModel(delegate: MockDelegate(), repository: mockRepository)
+
```
Comment (erikdvt) [inline Mooov/MooovTests/HomeTests/HomeTests.swift:19]: You have an extra new line here.

File: Mooov/MooovTests/MockRepository.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+//
+//  Created by Sinothando Mabhena on 2022/04/06.
+//
+
+import Foundation
+
+class MockRepository: MovieRepositoryType {
+    let movie: MovieInfo = MovieInfo(title: "Superman", year: "2020", rated: "8/10", released: "2020", runtime: "127min", genre: "action", director: "Zac Snyder", writer: "Zac Synder", actors: "Some one", plot: "A plot", language: "English", country: "USA", awards: "None", poster: "Poster", metascore: "4/5", imdbRating: "9/10", imdbVotes: "352", imdbId: "t12341234", type: "Movie", dvd: "dvd", boxOffice: "boxoffice", production: "prod", website: "website", response: "response", rating: [Rating(source: "source", value: "value")])
+    
+    let result: SearchResults = SearchResults(search: [Movie(title: "Batman", year: "2022", imdbId: "9/10", type: "Movie", poster: "Poster")], totalResults: "1", response: "true")
```
Comment (erikdvt) [inline Mooov/MooovTests/MockRepository.swift:13]: Maybe you could improve the readability of this? You could store it in a mock JSON file or just split it over multiple lines.

File: Mooov/Mooov/Home/Custom Cells/CollectionViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//  Mooov
+//
+//  Created by Sinothando Mabhena on 2022/04/04.
+//
+
+import Foundation
+import UIKit
+
+class CustomCollectionViewCell: UICollectionViewCell {
+    @IBOutlet weak var movieImageView: UIImageView!
```
Comment (bgouws) [inline Mooov/Mooov/Home/Custom Cells/CollectionViewCell.swift:12]: These can be `private`

File: Mooov/Mooov/Home/HomePageViewController.swift [hunk 4]
```diff
@@ -8,8 +8,49 @@
 import UIKit
 
 class HomePageViewController: UIViewController {
+    @IBOutlet weak var collectionView: UICollectionView!
```
Comment (bgouws) [inline Mooov/Mooov/Home/HomePageViewController.swift:11]: This can be `private` 😄

File: Mooov/Mooov/Home/HomePageViewController.swift [hunk 5]
```diff
@@ -8,8 +8,49 @@
...
+    
+    private lazy var viewModel = HomePageViewModel(delegate: self, repository: MovieRepository())
```
Comment (bgouws) [inline Mooov/Mooov/Home/HomePageViewController.swift:13]: Just a code style tip but you can put each parameter on its own line.

File: Mooov/Mooov/Home/HomePageViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    
+    func movie(atIndex: Int) -> MovieInfo? {
+        return fetchedMovies?[atIndex]
+    }
+    
+    func fetchMovies() {
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+                
+        repository?.fetchMovie("tt1877830", completion: { [weak self] result in
```
Comment (bgouws) [inline Mooov/Mooov/Home/HomePageViewModel.swift:39]: Are these movies hardcoded for now?

File: Mooov/MooovTests/MockRepository.swift (hunk 2 above)
Comment (bgouws) [reply Mooov/MooovTests/MockRepository.swift:13]: Each parameter on it's own line will be fine 😄

File: Mooov/MooovTests/HomeTests/MockDelegate.swift [hunk 7]
```diff
@@ -0,0 +1,13 @@
+//
+//  MockDelegate.swift
+//  Mooov
+//
+//  Created by Sinothando Mabhena on 2022/04/06.
+//
+
+import Foundation
+
+class MockDelegate: HomePageViewModelDelegate {
```
Comment (bgouws) [inline Mooov/MooovTests/HomeTests/MockDelegate.swift:10]: You can actually put this `MockDelegate` `class` inside `class HomeTests: XCTestCase {`

File: Mooov/MooovTests/MockRepository.swift [hunk 8]
```diff
@@ -0,0 +1,22 @@
+//
+//  MockRepository.swift
+//  Mooov
+//
+//  Created by Sinothando Mabhena on 2022/04/06.
+//
+
+import Foundation
+
+class MockRepository: MovieRepositoryType {
```
Comment (bgouws) [inline Mooov/MooovTests/MockRepository.swift:10]: This could also live inside of `class HomeTests: XCTestCase {` 😄

**Review by bgouws — CHANGES_REQUESTED** [review]
Looks pretty good, just a few enhancements 😄

File: Mooov/Mooov/Home/HomePageViewModel.swift (hunk 6 above)
Comment (smabhena) [reply Mooov/Mooov/Home/HomePageViewModel.swift:39]: Yes, the problem is my API only has two endpoints (get by id and get by search) so when it comes to a feature like this it doesn't have a new releases endpoint or an endpoint where i can get a bunch of movies from a certain date so it has to be hardcoded.

**Review by kirsty14 — DISMISSED** [review]
Looking good 👍🏻

File: Mooov/Mooov/Home/HomePageViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    
+    var count: Int {
+        return fetchedMovies?.count ?? 0
+    }
+    
+    func movie(atIndex: Int) -> MovieInfo? {
+        return fetchedMovies?[atIndex]
+    }
+    
+    func fetchMovies() {
```
Comment (bgouws) [inline Mooov/Mooov/Home/HomePageViewModel.swift:35]: So you should be able to test this too 😄 let me know if you need help 👍

File: Mooov/MooovTests/HomeTests/HomeTests.swift [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    
+    func testFetchMovie() {
+        viewModel.fetchMovies()
+        XCTAssertEqual(viewModel.movie(atIndex: 0)?.title, "Superman")
+    }
+    
+    func testMoviesCount() {
+        XCTAssertEqual(viewModel.count, 4)
+    }
+}
```
Comment (bgouws) [inline Mooov/MooovTests/HomeTests/HomeTests.swift:29]: Tests are looking great! lets see if we can get that last function tested 😄

**Review by bgouws — CHANGES_REQUESTED** [review]

File: Mooov/MooovTests/HomeTests/HomeTests.swift [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    func testMoviesCount() {
+        XCTAssertEqual(viewModel.count, 4)
+    }
+    
+    func testMovie() {
+        XCTAssertEqual(viewModel.movie(atIndex: 0)?.title, "Superman")
+    }
+}
+
+class MockDelegate: HomePageViewModelDelegate {
```
Comment (bgouws) [inline Mooov/MooovTests/HomeTests/HomeTests.swift:43]: Might be better to encapsulate these mock classes in `HomeTests` that way when you write tests for another class you can reuse the same naming 😄

**Review by bgouws — CHANGES_REQUESTED** [review]

File: Mooov/MooovTests/HomeTests/HomeTests.swift (hunk 11 above)
Comment (smabhena) [reply Mooov/MooovTests/HomeTests/HomeTests.swift:43]: Okay on it

**Review by kirsty14 — DISMISSED** [review]
Looking good 👍🏻

**Review by bgouws — APPROVED** [review]
Looks good 👍

**Pull request merged**